src/main/java/com/ait/lienzo/client/core/shape/decorator/IShapeDecorator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape.decorator;
+
+import com.ait.lienzo.client.core.shape.Shape;
+
+public interface IShapeDecorator<S extends Shape<?>> {
```

Comment:
Can you provide some java doc for this interface? I think it is not clear what is the purpose, especially what shape state is for.

Reply:
Sure, I'll provide some javadoc. The idea of this interface is that you can decorate a shape according to some State. I preferred to do that instead of hardcoding the styling of the shape as it was done before when the shape was being created. In this way, we can inject any custom decorator to the class that will be used to apply the style according to the given state.